Library/Homebrew/livecheck/options.rb

```diff
@@ -0,0 +1,105 @@
+# typed: strong
+# frozen_string_literal: true
+
+module Homebrew
+  module Livecheck
+    # Options to modify livecheck's behavior. These primarily come from
+    # `livecheck` blocks but they can also be set by livecheck at runtime.
+    #
+    # Option values use a `nil` default to indicate that the value has not been
+    # set.
+    class Options < T::Struct
+      # Whether to use brewed curl.
+      prop :homebrew_curl, T.nilable(T::Boolean)
+
+      # Form data to use when making a `POST` request.
+      prop :post_form, T.nilable(T::Hash[T.any(String, Symbol), String])
+
+      # JSON data to use when making a `POST` request.
+      prop :post_json, T.nilable(T::Hash[T.any(String, Symbol), String])
```

Comment:
Might be nicer to pick either a String or Symbol here and ensure callers are consistent/converted.

Reply:
Ha, good timing! I raised that issue yesterday on Slack and we generally agreed that we could standardize on symbol keys. I initially opted for strings because we may end up with field names that would cause problems for symbols (e.g., `one-two`, `1two`) but then I remembered that we can just use quote symbols (e.g., `:"one-two"`, `:"1two"`, `:"#{one}-two"`) to borrow the flexibility of strings.

I'm planning to create a PR to update the types and related documentation after `Options`.